Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM golang:1.17.6 as build
+
+WORKDIR /go/src/app
+ENV CGO_ENABLED=0
+ADD . /go/src/app
+
+RUN go get -d -v ./...
```

Comment:
I think that build will automatically pull in modules that don't exist locally. Nothing wrong in having it, but I don't think that you need it.

Reply:
Somewhat confusingly, what this command does is download the dependencies to the build image stage of the docker build. Only the binary gets copied to the actual output of the build.

I must confess that I cargo-culted this change from a previous repo, and having now looked at what it does, it seems it updates the `go.mod` and `go.sum` files to move any non-fixed dependencies to the latest available. This might be a good thing to do, but it also results in non-reproducible builds, so I'll get rid of it.